mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,277 @@
+	reqMap["request.host"] = req.GetAttributes().GetRequest().GetHttp().GetHost()
+	reqMap["request.method"] = req.GetAttributes().GetRequest().GetHttp().GetMethod()
+	reqMap["request.path"] = req.GetAttributes().GetRequest().GetHttp().GetPath()
+	reqMap["request.scheme"] = req.GetAttributes().GetRequest().GetHttp().GetScheme()
+	reqMap["request.time"] = reformatTime(req.GetAttributes().GetRequest().GetTime(), 0)
+	reqMap["request.useragent"] = req.GetAttributes().GetRequest().GetHttp().GetHeaders()["user-agent"]
+
+	pb.reqMap = reqMap
+
+	scope.Debugf("Returning bag with attributes:\n%v", pb)
+
+	return pb
+}
+
+// GetProtoBag returns a proto-based attribute bag.
+// When you are done using the proto bag, call the Done method to recycle it.
+//num is the index of the entry from the message's batch to create a bag from
+func GetEnvoyProtoBagAccessLog(msg *accessLogGRPC.StreamAccessLogsMessage, num int) attr.Bag {
+	// build the message-level dictionary
+	pb := envoyProtoBags.Get().(*EnvoyProtoBag)
+	reqMap := make(map[string]interface{})
+	if httpLogs := msg.GetHttpLogs(); httpLogs != nil {
+		reqMap["context.protocol"] = "http"
+		reqMap["context.reporter.kind"] = "inbound"
+		fillAddress(reqMap, httpLogs.GetLogEntry()[num].GetCommonProperties().GetDownstreamLocalAddress(),
+			"destination")
+		fillAddress(reqMap, httpLogs.GetLogEntry()[num].GetCommonProperties().GetDownstreamRemoteAddress(), "source")
+		reqMap["request.headers"] = attr.WrapStringMap(httpLogs.GetLogEntry()[num].GetRequest().GetRequestHeaders())
+		reqMap["request.host"] = httpLogs.GetLogEntry()[num].GetRequest().GetAuthority()
+		reqMap["request.method"] = httpLogs.GetLogEntry()[num].GetRequest().GetRequestMethod().String()
+		reqMap["request.path"] = httpLogs.GetLogEntry()[num].GetRequest().GetPath()
+		reqMap["request.scheme"] = httpLogs.GetLogEntry()[num].GetRequest().GetScheme()
+		reqMap["request.time"] = reformatTime(httpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime(), 0)
+		//is this the right time
+		reqMap["response.time"] = reformatTime(httpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime(),
+			httpLogs.GetLogEntry()[num].GetCommonProperties().GetTimeToFirstUpstreamRxByte().Nanos)
+		reqMap["request.useragent"] = httpLogs.GetLogEntry()[num].GetRequest().GetUserAgent()
+		reqMap["response.headers"] = attr.WrapStringMap(httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeaders())
+		reqMap["response.code"] = int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseCode().GetValue())
+		reqMap["response.size"] = int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseBodyBytes())
+		reqMap["response.total_size"] = int64(httpLogs.GetLogEntry()[num].GetResponse().
+			GetResponseBodyBytes()) + int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeadersBytes())
+	} else if tcpLogs := msg.GetTcpLogs(); tcpLogs != nil {
+		reqMap["context.protocol"] = "tcp"
```

Comment:
Connection property has sent/received bytes, which should also be mapped to connection.bytes_received/sent.